spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteDataFilesSparkAction.java

```diff
@@ -0,0 +1,443 @@
+
+    return new Result(results.values().stream().collect(Collectors.toMap(Pair::first, Pair::second)));
+  }
+
+  private Result doExecutePartialProgress(Stream<Pair<FileGroupInfo, List<FileScanTask>>> groupStream,
+      RewriteStrategy strategy, int totalGroups, Map<StructLike, Integer> numGroupsPerPartition) {
+
+    ExecutorService rewriteService = Executors.newFixedThreadPool(maxConcurrentFileGroupActions,
+        new ThreadFactoryBuilder().setNameFormat("Rewrite-Service-%d").build());
+
+    ExecutorService committerService = Executors.newSingleThreadExecutor(
+        new ThreadFactoryBuilder().setNameFormat("Committer-Service").build());
+
+    int groupsPerCommit = partialProgressEnabled ?
+        IntMath.divide(totalGroups, maxCommits, RoundingMode.CEILING) :
+        totalGroups;
+
+    AtomicBoolean stillRewriting = new AtomicBoolean(true);
+    ConcurrentHashMap<String, Pair<FileGroupInfo, FileGroupRewriteResult>> results = new ConcurrentHashMap<>();
+    ConcurrentLinkedQueue<String> completedRewriteIds = new ConcurrentLinkedQueue<>();
+    ConcurrentLinkedQueue<String> completedCommitIds = new ConcurrentLinkedQueue<>();
+
+    // Partial progress commit service
+    committerService.execute(() -> {
+      while (stillRewriting.get() || completedRewriteIds.size() > 0) {
+        Thread.yield();
+        // Either we have a full commit group, or we have completed writing and need to commit what is left over
+        if (completedRewriteIds.size() > groupsPerCommit ||
+            (!stillRewriting.get() && completedRewriteIds.size() > 0)) {
+
+          Set<String> batch = Sets.newHashSetWithExpectedSize(groupsPerCommit);
+          for (int i = 0; i < groupsPerCommit && !completedRewriteIds.isEmpty(); i++) {
+            batch.add(completedRewriteIds.poll());
+          }
+
+          try {
+            commitOrClean(batch);
+            completedCommitIds.addAll(batch);
+          } catch (Exception e) {
+            batch.forEach(results::remove);
+            LOG.error("Failure during rewrite commit process, partial progress enabled. Ignoring", e);
+          }
+        }
+      }
+    });
+
+    // Start rewrite tasks
+    Tasks.foreach(groupStream.iterator())
+        .suppressFailureWhenFinished()
+        .executeWith(rewriteService)
+        .noRetry()
+        .onFailure((info, exception) -> {
+          LOG.error("Failure during rewrite process for group {}", info.first(), exception);
+          abortFileGroup(info.first().groupID);
+        })
+        .run(infoListPair ->
+            rewriteFiles(infoListPair, totalGroups, numGroupsPerPartition, strategy, completedRewriteIds, results));
+
+    stillRewriting.set(false);
+    committerService.shutdown();
+
+    try {
+      committerService.awaitTermination(10, TimeUnit.MINUTES);
+    } catch (InterruptedException e) {
+      throw new RuntimeException("Cannot complete commit for rewrite, commit service interuptted", e);
```

Comment:
typo: `interrupted`?

Reply:
Yep sorry didn't push that fix yet, I noticed this on my last run through